lib/Charts/Title.js

```diff
@@ -70,20 +73,20 @@ const Title = {
             // Enter.
             const addedTitleComponents = titleComponents.enter().append('span')
                 .attr('class', 'title-component');
+
             if (titleSettings.type === 'legend') {
                 addedTitleComponents.append('span')
                     .attr('class', 'color');
             }
             addedTitleComponents.append('span')
                 .attr('class', 'name');
             // Enter and update.
-            titleComponents.select('.color').style('background-color', d=>d.color);
-            titleComponents.select('.name').text(getName);
+            addedTitleComponents.select('.color').style('background-color', d=>d.color);
+            addedTitleComponents.select('.name').text(getName);
```

Comment:
I can't remember why I had `titleComponents` here instead of `addedTitleComponents` - what made you switch?